### ISSUE TITLE
Consistency (and improved naming?) for `llvm` properties

### ISSUE THREAD

**Issue by dabrahams** [hylo-lang/Swifty-LLVM#48]
In order to avoid exposing `llvmc` to clients, #47 wrapped all `public` properties named `llvm`, whose type was defined in `llvmc`, in `public` `struct`s, with the underlying value being accessed via a `.raw` property.  This leaves us with many private/internal properties named `llvm` that have no `.raw` property, which makes usage a bit confusing.  With some thinking a better arrangement is likely possible.  We could probably benefit by putting even the private/internal properties in wrappers, because right now they are all `OpaquePointer` instances and the wrappers add some static type safety.

The names `llvm` and `raw` are not great either, so it would be good to adjust those if possible.

**Comment by kyouko-taiga** [comment]
I agree that `llvm` isn't a great name. Initially I thought about `handle` but that's not right either. LLVM's C API reimplements some of C++'s enum in C and those are exposed to Swift directly. Therefore some of the properties that are called `llvm` in our code base aren't opaque pointers but simple values that we could read and update in Swift directly.

Perhaps we could use `representationInLLVM`?

We don't necessarily need a unique name for everything but the original design goal was to have a uniform way to talk about the value that is being passed to interact with LLVM API, regardless of what it is. For a `llvm::Module` it's a handler implemented as an opaque pointer and for `llvm::Reloc::Model` it's some C enum. In both cases, this value should be irrelevant the the user of SwiftyLLVM because we should offer a consistent Swift API for everything.

There is _some_ value in having access to these handlers and enum cases for debugging and hot fixing so I'm not sure they should be completely inaccessible.

**Comment by dabrahams** [comment]
> Perhaps we could use representationInLLVM

In what sense is that thing a representation "in LLVM?"  

My first assumption from that name would be that the parameter is a raw LLVM handle, which I think(?) it is not.  But even if it is, “in LLVM” doesn't quite make sense to me.

s/handler/handle/g I assume.

**Comment by tothambrus11** [comment]
Users of the library are not supposed to use the `.llvm` property, it's a library-internal detail required for protocols and protocol-specific implementations to work. I couldn't find a more suitable name, but feel free to reopen if you still find this actionable.
